### ISSUE TITLE
New compiler: chained access expressions broken

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#2610]
CC @fernewelten 

Following kind of expressions appear broken after the recent PR #2583:

```
int id = Hotspot.GetAtScreenXY(0, 0).ID;
```

Compiler reports error:
> Cannot assign a type 'Hotspot *' value to a type 'int' variable

Another case:
```
import String Func();

function game_start()
{
    int x = Func().AsInt;
}
```

> Cannot assign a type 'String' value to a type 'int' variable

We should also add this syntax case into the compiler tests.

**Comment by fernewelten** [comment]
OK, I'll have a look.

**Comment by ivan-mogilko** [comment]
Fixed by #2611

**Comment by ivan-mogilko** [comment]
@fernewelten I'm sorry, i rushed merging the pr and closing this.

There's still expressions that have same problem:
```
import String Func();

function game_start()
{
    int x = Func().AsInt;
}
```

I do not know what's the difference between this and the previously reported one.

**Comment by fernewelten** [comment]
Okay, I'll see to it in the morning.

**Comment by ivan-mogilko** [comment]
Finally fixed by #2614
